sdk/appplatform/mgmt/src/test/java/com/azure/resourcemanager/appplatform/SpringCloudTest.java

```diff
@@ -0,0 +1,125 @@
+            .withSku("S0")
+            .apply();
+
+        Assertions.assertEquals("S0", service.sku().name());
+
+        service.update()
+            .withGitUri("https://github.com/Azure-Samples/piggymetrics-config")
+            .apply();
+        Assertions.assertEquals("https://github.com/Azure-Samples/piggymetrics-config", service.serverProperties().configServer().gitProperty().uri());
+
+        File jarFile = new File("gateway.jar");
+        if (!jarFile.exists()) {
+            URLConnection connection = new URL("https://github.com/weidongxu-microsoft/azure-sdk-for-java-management-tests/raw/master/spring-cloud/gateway.jar").openConnection();
+            connection.connect();
+            IOUtils.copy(connection.getInputStream(), new FileOutputStream(jarFile));
+        }
+
+        SpringApp app = service.apps().define(appName)
+            .withPublicEndpoint()
+            .create();
+
+        Assertions.assertNotNull(app.url());
+        Assertions.assertNotNull(app.activeDeployment());
+
+        Assertions.assertTrue(requestSuccess(app.url()));
+
+        app.update()
+            .withoutDeployment(app.activeDeployment())
+            .deployJar(deploymentName, jarFile)
+            .apply();
+
+        Assertions.assertNotNull(app.url());
+        Assertions.assertEquals(deploymentName, app.activeDeployment());
+
+        Assertions.assertTrue(requestSuccess(app.url()));
+
+        SpringAppDeployment deployment = app.deployments().getByName(app.activeDeployment());
+        deployment.update()
+            .withCpu(2)
+            .withMemory(4)
+            .withRuntime(RuntimeVersion.JAVA_11)
+            .withInstance(2)
+            .apply();
+
+        Assertions.assertEquals(2, deployment.settings().cpu());
+        Assertions.assertEquals(4, deployment.settings().memoryInGB());
+        Assertions.assertEquals(RuntimeVersion.JAVA_11, deployment.settings().runtimeVersion());
+        Assertions.assertEquals(2, deployment.instances().size());
+
+        deployment = app.deployments().define(deploymentName1)
+            .withSourceCodeFolder(new File(this.getClass().getResource("/piggymetrics").getFile()))
+            .withTargetModule("gateway")
+            .withSettingsFromActiveDeployment()
+            .activate()
+            .create();
+        app.refresh();
+
+        Assertions.assertEquals(deploymentName1, app.activeDeployment());
+        Assertions.assertEquals(2, deployment.settings().cpu());
+        Assertions.assertNotNull(deployment.getLogFileUrl());
+
+        Assertions.assertTrue(requestSuccess(app.url()));
+
+        app.update()
+            .withoutPublicEndpoint()
+            .apply();
+        Assertions.assertFalse(app.isPublic());
+    }
+
+    private boolean requestSuccess(String url) throws IOException {
+        for (int i = 0; i < 60; ++i) {
+            HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+            connection.connect();
```

Comment:
Need to close it somewhere.

https://docs.oracle.com/javase/7/docs/api/java/net/HttpURLConnection.html#disconnect()